client/cli/src/commands/runcmd.rs

```diff
@@ -562,13 +563,34 @@ fn interface_str(
 	}
 }
 
-/// Default to verbosity level 0, if none is provided.
+/// Utility error for returning custom errors that implement the std::error::Error trait.
+#[derive(Debug)]
+struct TelemetryParsingError {
+    details: String
+}
+
+impl TelemetryParsingError {
+    fn new(msg: &str) -> TelemetryParsingError {
+        TelemetryParsingError{details: msg.to_string()}
+    }
+}
+
+impl fmt::Display for TelemetryParsingError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.details)
+    }
+}
+
+impl Error for TelemetryParsingError {
+    fn description(&self) -> &str {
+        &self.details
+    }
+}
+
 fn parse_telemetry_endpoints(s: &str) -> Result<(String, u8), Box<dyn std::error::Error>> {
```

Comment:
```suggestion
fn parse_telemetry_endpoints(s: &str) -> Result<(String, u8), TelemetryParsingError> {
```

Reply:
I'm missing something here: what should we do with the result return by the `parse()` function ? Should I add something to the `TelemetryParsingError` enum?

Reply:
> Should I add something to the TelemetryParsingError enum?

Yes!

`TelemetryParsingError` by definition is an error that can happen when parsing an address, and should thus contain all the possible problems that can happen.